packages/peregrine/lib/talons/SearchBar/useAutocomplete.js

```diff
@@ -23,7 +24,24 @@ export const useAutocomplete = props => {
     const { value } = useFieldState('search_query');
     const valid = value && value.length > 2;
 
-    // determine message type
+    // Create a debounced function so we only search some delay after the last
+    // keypress.
+    const debouncedRunQuery = useMemo(
+        () =>
```

Comment:
Why not just debounce the callback to useMemo? Did you try it and it was weird? It seems like it should work out of the box.

```jsx
const debouncedRunQuery = useMemo(
  debounce(runQuery, 500),
  [runQuery]
);
```

Reply:
I tried this and it didn't work :/ I can try again, maybe I borked the syntax or something.

Reply:
Yea I can't do this -- I can simplify and not forward `opts` in the internal callback to `debounce` but `useMemo` requires a ["create" function](https://reactjs.org/docs/hooks-reference.html#usememo).

Reply:
Oh, of course. My bad.